src/components/ModalInput/ModalInput.tsx

```diff
@@ -68,15 +70,26 @@ const ModalInput: React.FC<ModalInputProps> = ({
   const TranslateString = useI18n()
   const isBalanceZero = max === '0' || !max
 
-  const displayBalance = isBalanceZero ? '0' : parseFloat(max).toFixed(4)
+  const displayBalance = (balance: string) => {
+    if (isBalanceZero) {
+      return '0'
+    }
+    const balanceNumber = new BigNumber(balance)
+    const balanceNumberFixed = balanceNumber.toFixed(4)
```

Comment:
This seems a bit unnecessary. Why not just use `toLocalString` for both cases?
```
return balance.toLocalString(undefined, { maximumFractionDigits: 4 });
```

Reply:
Done